Foundation/NSStream.swift

```diff
@@ -209,20 +227,32 @@ public class NSOutputStream : Stream {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
+        guard let property = property else { return false }
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
 }
 
 // Discussion of this API is ongoing for its usage of AutoreleasingUnsafeMutablePointer
 #if false
```

Comment:
@phausler:

It doesn't have to be a pointer, could we return a tuple? `-> (input: InputStream, output: OutputStream)`.

I recognize the API won't be final, but just something to allow this behavior since all the pieces are there. Would this work (or are the underlying reqs not available):

```Swift
    public class func getStreamsToHost(withName hostname: String, port: Int) -> (input: InputStream, output: OutputStream)? {
        var read: Unmanaged<CFReadStream>?
        var write: Unmanaged<CFWriteStream>?
        CFStreamCreatePairWithSocketToHost(kCFAllocatorDefault, hostname._cfObject, UInt32(port), &read, &write)
        guard let input = read?.takeRetainedValue(), output = write?.takeRetainedValue() else { return nil }
        return (input, output)
    }

```